### PR TITLE
Fix fire exploit being extinguished by pAIs, borgs, and simple animals

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

This fixes a portion of #48926.  pAIs, borgs, and simple mobs can no longer walk over a mob on fire and put them out.  

## Why It's Good For The Game

Less powergaming exploits.

## Changelog
:cl:
fix: Fix fire being extinguished by pAIs, borgs, or simple animals by walking over the mob.
/:cl:

### CODE REVIEW

**Comment by timothymtorres** [comment]
Going to rework this to be a trait for certain mobs.

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -1442,6 +1442,9 @@
 	if(!istype(L))
 		return
 
+	if(issilicon(L) || issilicon(src) || isanimal(L) || isanimal(src))
+		return // can't spread fire to mobs that don't catch on fire
```
Comment (ATH1909) [inline code/modules/mob/living/living.dm:1446]: can't silicons catch on fire, though?

Comment (timothymtorres) [reply code/modules/mob/living/living.dm:1446]: Borgs can catch fire but it’s only for aesthetics.  They take no fire damage.  If I remember correctly, it only happens during extreme heat for the SM.

**Comment by timothymtorres** [comment]
It's ready now.

**Review by spookydonut — APPROVED** [review]
good trait usage

**Pull request merged**